src/classes/frDonation.cls

```diff
@@ -53,84 +54,136 @@ public class frDonation extends frModel implements frSyncable {
         return mappings;
     }
     
-    private Opportunity o;
+    private SObject o;
     
     public frDonation(Sync_Attempt__c syncRecord){
         super(syncRecord);
     }
-    
+
     public Boolean sync() {
+        //create Sobject
         Boolean result = false;
         Map<String, Object> request = getRequestBody();
-        o = new Opportunity();
+        for (String key : request.keySet()) {
+            System.debug('Key: ' + key + ', Value: ' + request.get(key));
+        }
+        String frId = getFunraiseId();
+
+        if(!frUtil.hasNPCobjects()) {
+            o = new Opportunity();
+            applyMappings(o, request);
+            setOpportunityStage((Opportunity) o, String.valueOf(request.get('status')));
+        } 
+        else {
+            List<SObject> gtList = Database.query('SELECT Id, Name, DonorId, AcknowledgementDate FROM GiftTransaction WHERE fr_ID__c = :frId');
+            if(gtList.isEmpty()) {
+                o = Schema.getGlobalDescribe().get('gifttransaction').newSObject();
+            } 
+            else {
+                o = gtList.get(0);
+            }
+            o.put('Name', String.valueOf(request.get('name')));
+            o.put('OriginalAmount', (Decimal) request.get('amount'));
+            o.put('PaymentMethod', String.valueOf(request.get('paymentMethodType')));
+            o.put('AcknowledgementDate', System.today());
+        }
+
         applyMappings(o, request);
-        setOpportunityStage(o, String.valueOf(request.get('status')));
-        
         String funraiseId = getFunraiseId();       
-        String supporterFunraiseId = String.valueOf(request.get('donorId'));        
+        String supporterFunraiseId = String.valueOf(request.get('donorId')); 
+        String supporterObjName = frUtil.hasNPCobjects() ? 'Account' : 'Contact';
+        String subsciptionObjName = frUtil.hasNPCobjects() ? 'GiftCommitment' : 'Subscription__c';
+        
+        //connect supporter
         if (String.isNotBlank(supporterFunraiseId)) {
-            List<Contact> contacts = [SELECT Id, AccountId FROM Contact WHERE fr_ID__c = :supporterFunraiseId];
-            if (contacts.size() > 0) {
-                Contact supporterContact = contacts.get(0);
-                o.fr_Donor__c = supporterContact.Id;
+            List<SObject> supporters;
+            if(frUtil.hasNPCobjects()) supporters = Database.query('SELECT Id FROM ' + supporterObjName + ' WHERE fr_ID__c = :supporterFunraiseId');
+            else supporters = [SELECT Id, AccountId FROM Contact WHERE fr_ID__c = :supporterFunraiseId];
+            if (supporters.size() > 0) {
+                SObject supporter = supporters.get(0);
+                o.put('DonorId', supporter.get('Id'));
```

Comment:
I don't think this field will exist in the non-NPC case, as there isn't a standard field with this name on the Opportunity object.  The removed line 75 in this case is a key piece of functionality where the Opportunity and Contact are linked

Reply:
Changed to use the correct field in non NPC orgs!